src/js/controllers/tab-receive.js

```diff
@@ -88,9 +82,9 @@ angular.module('copayApp.controllers').controller('tabReceiveController', functi
     });
   };
 
-  $scope.setWallet = function(index) {
-    $scope.wallet = $scope.wallets[index];
-    $scope.walletIndex = index;
+  $scope.setWallet = function(wallet) {
+    $scope.wallet = wallet;
+    $scope.walletIndex = $scope.wallets.indexOf(wallet);
     if ($scope.walletAddrs[$scope.wallet.id].addr) $scope.addr = $scope.walletAddrs[$scope.walletIndex].addr;
```

Comment:
What would happen if `$scope.walletAddrs[$scope.wallet.id].addr` was `undefined`? I think you should reset `$scope.addr` first.